WooCommerce/Classes/ViewRelated/Dashboard/Reviews/ReviewsDashboardCardViewModel.swift

```diff
@@ -19,30 +19,185 @@ final class ReviewsDashboardCardViewModel: ObservableObject {
          analytics: Analytics = ServiceLocator.analytics) {
         self.siteID = siteID
         self.stores = stores
-        self.storage = storage
+        self.storageManager = storageManager
         self.analytics = analytics
+
+        self.productsResultsController = ResultsController<StorageProduct>(storageManager: storageManager,
+                                                                           matching: nil,
+                                                                           sortedBy: [])
+        configureProductReviewsResultsController()
     }
 
+    /// ResultsController for ProductReview
+    private lazy var productReviewsResultsController: ResultsController<StorageProductReview> = {
+        let sortDescriptor = NSSortDescriptor(keyPath: \StorageProductReview.dateCreated, ascending: false)
+        return ResultsController<StorageProductReview>(storageManager: storageManager,
+                                                       matching: sitePredicate(),
+                                                       fetchLimit: Constants.numberOfItems,
+                                                       sortedBy: [sortDescriptor])
+    }()
+
     func dismissReviews() {
         // TODO: add tracking
         onDismiss?()
     }
+
+    @MainActor
+    func reloadData() async {
+        syncingData = true
+        syncingError = nil
+        do {
+            // Ignoring the result from remote as we're using storage as the single source of truth
+            _ = try await loadReviews()
+        } catch {
+            syncingError = error
+        }
+        syncingData = false
+    }
+}
+
+// MARK: - Storage related
+private extension ReviewsDashboardCardViewModel {
+    /// Predicate to entities that belong to the current store
+    ///
+    func sitePredicate() -> NSPredicate {
+        return NSPredicate(format: "siteID == %lld", siteID)
+    }
+
+    /// Performs initial fetch from storage and updates results.
+    ///
+    func configureProductReviewsResultsController() {
+        productReviewsResultsController.onDidChangeContent = { [weak self] in
+            guard let self else { return }
+            Task {
+                await self.updateReviewsResults()
+            }
+        }
+        productReviewsResultsController.onDidResetContent = { [weak self] in
+            guard let self else { return }
+            Task {
+                await self.updateReviewsResults()
+            }
+        }
+
+        do {
+            try productReviewsResultsController.performFetch()
+        } catch {
+            ServiceLocator.crashLogging.logError(error)
+        }
+    }
+
+    func updateProductsResultsControllerPredicate(with productIDs: [Int64]) {
+        let predicates = NSCompoundPredicate(andPredicateWithSubpredicates: [sitePredicate(),
+                                                                            NSPredicate(format: "productID IN %@", productIDs)])
```

Comment:
Super nit: Indentation fix needed.

Reply:
I think this looks odd but perhaps correct because it's an array as the parameter.

Either using enter manually or Editor > Structure > Re-indent results the same like above:

https://github.com/woocommerce/woocommerce-ios/assets/266376/445f0afc-45a6-484d-b386-0714cc9874c9